### ISSUE TITLE
Code suggestions

### ISSUE THREAD

**Issue by jklymak** [kurtisanstey/project#25]
- Just leave the dates as datetime64 and don't mix pandas.  `np.datetime64('2018-01-01')` is just as easy (or easier) than `datetime(2018,1,1)`

I don't think you need to go from pandas and back just to linearly interpolate...  `np.interp` should work .

- we already discussed the output of the co-ords in the netcdf files.  

- not clear why you have an `if/else` for the saving of the files....  

- I think its probably fine to just save your "segments" as separate files.  But those time series should concatenate if possible.  Another approach would've been to assign a segment number as a variable. Then you could have done 

```
ds = xr.load_dataset('boo.nc')
d = ds.isel(segment=i)
```
and d would just be the data that had the segment you are interested in.  The advantage of this, is that you have the data all in one place.  

### File 2

I think its OK to deal with lists of data sets like this, but consider the approach suggested above as well...

Building lists of arrays is a bit inelegant.  I'd instead do: 

```
um_PSD = np.zeros((nsegs, ndepths, nfs )) 
um_PSD[i, j, :] = sig.welch()....
```

But its not terrible to do what you are doing...

OK, you can just save N2 as a variable with just depth dimension, and multiply when needed during plotting.  You don't need to divide by "int_scale" and save that.  Its just a multiplicative factor.  Regardless, you *definitely* want to save that factor.  

Again, not your only dims here are f_PSD and depth 

*MAJOR* I'm actually confused here.  I thought the plan was for you to make matrices of raw spectral estimates, and then average them as need be. i.e. *don't* average, and hence have many many segments, and save as 3-D matrices, with dimensions `depth`, `time`, `f`, where time is the beginning or middle of each time slice, and each time slice represents 1024 15-minute averages (or whatever NFFT is).  i.e. for a year it would be a matrix of 220, 30, 1024.   spectra that are gaps, are just left as NaN.  Then when you want to plot the average, you need only select the time interval to average (ignoring NaN).   Or if you are plotting a spectrogram, you simply pcolor the spectrogram at the depth of interesting.  

Or is that coming later?  It really should be here.

**Comment by kurtisanstey** [comment]
@jklymak 

Hi Jody, not sure you saw my other Issue thread. I updated the code from my work today after you posted this. The depth-time-frequency matrices come from my spectrogram process which I found useful to produce that data, and is the last step I'm working on (the ds.dataset is still having trouble associating three coordinates to a variable), so what you were looking at today is both slightly out-of-date, and missing the depth-time-frequency process. The PSD and rotary spectra here are from my old averaging process, which was useful for making the sure the iterative rework was working properly. I'll let you know when I've addressed your comments and finished with the 3D matrices.
